foundations/javascript_basics/function_basics.md

```diff
@@ -80,6 +80,8 @@ The following questions are an opportunity to reflect on key topics in this less
 - [What is function scope?](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Functions#function_scope_and_conflicts)
 - [What are return values?](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Return_values#what_are_return_values)
 - [What are arrow functions?](https://javascript.info/arrow-functions-basics)
+- [What is a function expression?](https://javascript.info/function-expressions)
```

Comment:
Arrow functions are a type of function expression, just a different syntax to the traditional syntax (plus some other small caveats). Given the next proposed question which covers declarations vs expressions, I think this particular one is a little redundant.
```suggestion
```

Reply:
"Arrow functions are a type of function expression"
I thought they were different, my bad!

I’m not sure why you have found the second question about `the difference between a function declaration and a function expression` redundant. There is one big difference that led me to add this question: a function declaration gets hoisted and can be accessed before its declaration line. On the other hand, if you try to access a function expression before its declaration, you will get an error. I thought adding a question about this would help it stick in learners' minds.

I totally agree that it might seem trivial, but it’s just a question about something we’ve just learned. For example, look at the other existing questions in [knowledge-check](https://www.theodinproject.com/lessons/foundations-function-basics#knowledge-check):

- What are functions useful for?
- How do you invoke a function?

These are straightforward and simple questions, but I think they are important and serve their purpose.

Reply:
I think you've misunderstood which question I referred to as redundant. The one I highlighted for deletion ("What is a function expression?") is somewhat redundant since arrow functions are also a type of function expression, and I'm not sure the question adds a great deal. The second question you added is perfectly fine IMO.

On top of this, what Zach pointed out about the PR form is correct - you need to explain why your made the proposed changes as opposed to just making them and saying what they are.

Reply:
Thanks @MaoShizhong, next time I'll be sure to fill all the sections of the form.

Reply:
@youssef-el-atmani To clarify in case you misunderstood - I'm requesting that the first knowledge check question you added is removed. We only need the second question you added. Once the first one is removed, I'm happy to approve and merge this PR.

Reply:
Sorry @MaoShizhong , I did get that the first  question should be removed, but I didn't get that I am the one who should remove it!
**Done**: First question has been removed

Thank you a lot for the clarification, and sorry because this PR toked way more time than it needed!